Marlin/src/module/tool_change.cpp

```diff
@@ -495,14 +499,28 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
 
           #if HAS_SOFTWARE_ENDSTOPS
             // Update the X software endstops early
+            const float old_extruder = active_extruder;
             active_extruder = tmp_extruder;
             update_software_endstops(X_AXIS);
-            active_extruder = !tmp_extruder;
+            active_extruder = old_extruder;
```

Comment:
Fine, but an unnecessary change. We know that the tool is definitely being changed at this point, and we know that 0 and 1 are the only valid values for `tmp_extruder`.

Reply:
For now, but i figured this may be more future-proof if anything adds more down the road.

Reply:
Compile size takes precedence. We can update it when it _needs_ to be updated.